### ISSUE TITLE
Cached gems layer reused even when ruby minor version changes

### ISSUE THREAD

**Issue by fg-j** [paketo-buildpacks/bundle-install#112]
## What happened?

We built a Ruby app with version `2.7.2` of mri, then rebuilt it, changing the version to `2.6.6`. The app failed to run after rebuild because it was not able to find the required gems.

## Reproduction Steps

1. Clone [this sample app](https://github.com/paketo-buildpacks/samples/tree/main/ruby/puma)
2. Generate a `Gemfile.lock` by running `bundle install` locally.
3. Add a `buildpack.yml` that specifies an MRI version:
```yaml
---

mri:
  version: 2.7.2
```
4. Build the app:
```shell
pack build changing-version --builder paketobuildpacks/builder:0.1.20-base --buildpack gcr.io/paketo-buildpacks/ruby:0.1.5 --clear-cache
```
5. Run the app
```
docker run -d --env PORT=8080 --publish 8080:8080 changing-version
```
6. And see the expected output when accessing it:
```
curl 0.0.0.0:8080
<!DOCTYPE html>
  <html>
    <head>
      <title>Powered By Paketo Buildpacks</title>
    </head>
    <body>
      <img style="display: block; margin-left: auto; margin-right: auto; width: 50%;" src="https://paketo.io/images/paketo-logo-full-color.png"></img>
    </body>
  </html>
```

7. Then, change the `buildpack.yml` to specify a different MRI version:
```yaml
---

mri:
  version: 2.6.6
```
8. Rebuild the app without clearing the cache:
```shell
 pack build changing-version --builder paketobuildpacks/builder:0.1.20-base --buildpack gcr.io/paketo-buildpacks/ruby:0.1.5
```

9. Note that build output includes:
```
[builder] Paketo Bundle Install Buildpack 0.1.0
[builder]   Reusing cached layer /layers/paketo-buildpacks_bundle-install/gems
[builder]
```
10. Re-run the app (in interactive mode); note the error output:
```
docker run -it --env PORT=8080 --publish 8080:8080 changing-version
bundler: command not found: puma
Install missing gem executables with `bundle install`
```
11. Re-run the app and jump into the container; note that the gems are in a `ruby/2.7.0` directory
```shell
docker run -it --env PORT=8080 --publish 8080:8080  --entrypoint=launcher changing-version  /bin/bash
cnb@80bec5a816a0:/workspace$ ll /layers/paketo-buildpacks_bundle-install/gems/ruby/
total 12
drwxr-xr-x 3 cnb cnb 4096 Jan  1  1980 ./
drwxr-xr-x 4 cnb cnb 4096 Jan  1  1980 ../
drwxr-xr-x 9 cnb cnb 4096 Jan  1  1980 2.7.0/
```
The buildpack should rebuild the gem layer when the minor version of ruby changes, because the gems will need to be placed in a different subdirectory.

## Checklist

Please confirm the following:

* [x] I have included log output.
* [x] The log output includes an error message.
* [x] I have included steps for reproduction.

**Comment by arjun024** [comment]
This is fixed by the [bundle-install v0.1.1](https://github.com/paketo-buildpacks/bundle-install/releases/tag/v0.1.1), which is available in the metabuildpack `ruby:0.1.6`. Please reopen it you feel otherwise.

**Comment by ryanmoran** [comment]
I do not think this is resolved by the fix that is mentioned. I followed the above reproduction criteria, with a minor additon (I also include a `Gemfile.lock` in the source code) and am still seeing the missing gem issue. The above fix would only resolve the issue if the user did not include a `Gemfile.lock`.

**Comment by fg-j** [comment]
Using the same reproduction steps as the original issue, this has now been resolved as of Ruby buildpack v0.4.0.
Running
```
pack build changing-version --buildpack gcr.io/paketo-buildpacks/ruby:0.4.0
```
after changing the version in the `buildpack.yml` shows that the cached gems layer is no longer being reused:
```
[builder] Paketo Bundler Buildpack 0.1.4
[builder]   Resolving Bundler version
[builder]     Candidate version sources (in priority order):
[builder]       <unknown> -> ""
[builder]
[builder]     Selected Bundler version (using <unknown>): 2.2.11
[builder]
[builder]   Reusing cached layer /layers/paketo-buildpacks_bundler/bundler
[builder]
[builder] Paketo Bundle Install Buildpack 0.1.6
[builder]   Executing build process
[builder]     Running 'bundle config path /layers/paketo-buildpacks_bundle-install/gems'
[builder]     Running 'bundle config clean true'
[builder]     Running 'bundle config cache_path --parseable'
[builder]     Running 'bundle install'
[builder]       Completed in 11.327s
[builder]
[builder]   Configuring environment
[builder]     BUNDLE_PATH -> "/layers/paketo-buildpacks_bundle-install/gems"
```
and `docker run` now succeeds on the rebuilt app.

Also, after rebuilding, if we look at the layer contents in the rebuilt image, we see:
```
docker run -it --env PORT=8080 --publish 8080:8080  --entrypoint=launcher changing-version  /bin/bash
cnb@15dcfbddbd87:/workspace$ ll /layers/paketo-buildpacks_bundle-install/gems/ruby/
total 12
drwxr-xr-x 3 cnb cnb 4096 Jan  1  1980 ./
drwxr-xr-x 4 cnb cnb 4096 Jan  1  1980 ../
drwxr-xr-x 9 cnb cnb 4096 Jan  1  1980 2.6.0/
```
meaning that the directory with the *stale* 2.7.0 gems has also been cleaned up.
